### DISCUSSION TITLE
Support hybrid S3 compatible storages

### DISCUSSION THREAD

**Discussion by linzhu-isoftstone** [awspring/spring-cloud-aws#188] [Ideas]
In some cases, the storage systems consist of both AWS S3 on cloud and S3 compatible on-premise OSS like MinIO. With the current design and implementation of Spring Cloud AWS, it's not possible to have MinIO and S3 configured in the same project. 

Is it possible that we add the support for multiple hybrid S3 compatible storage systems working together?

**Comment by maciejwalkowiak** [comment]
Apologies for late reply. At this point we do not have a plan for that, but if its something you would like to contribute and you have a proper design in mind we are happy to discuss.
